docs/src/man/basics.md

```diff
@@ -1565,604 +1565,1630 @@ julia> german[Not(5), r"S"]
+and returns the result of the vector addition.
+However, in that same line,
+we assigned the resultant `Vector` to a new column `z` in the data frame `df`.
+We could have instead assigned the resultant `Vector` to some other variable,
+and then `df` would not have been altered.
+The approach with dot syntax is very versatile
+since the data getting, mathematics, and data setting can be separate steps.
+
+```julia
+julia> df.x  # dot syntax returns a vector
+3-element Vector{Int64}:
+ 1
+ 2
+ 3
+
+julia> v = df.x + df.y  # assign mathematical result to a vector `v`
+3-element Vector{Int64}:
+ 5
+ 7
+ 9
+
+julia> df.z = v  # place `v` into the data frame `df` with the column name `z`
+3-element Vector{Int64}:
+ 5
+ 7
+ 9
+```
+
+One downside to dot syntax is that the column name must be explicitly written in the code.
+Indexing syntax can perform a similar operation with dynamic column names.
+(Manipulation functions can also work with dynamic column names as will be shown in the next example.)
+
+```julia
+julia> df = DataFrame("My First Column" => 1:3, "My Second Column" => 4:6)  # define data frame
+3×2 DataFrame
+ Row │ My First Column  My Second Column
+     │ Int64            Int64
+─────┼───────────────────────────────────
+   1 │               1                 4
+   2 │               2                 5
+   3 │               3                 6
+
+julia> c1 = "My First Column"; c2 = "My Second Column"; c3 = "My Third Column";  # define column names
+
+# Imagine the above data was read from a file or entered by a user at runtime.
+
+julia> df.c1  # dot syntax expects an explicit column name and cannot be used
+ERROR: ArgumentError: column name :c1 not found in the data frame
+
+julia> df[:, c3] = df[:, c1] + df[:, c2]  # access columns with names stored in variables
+3-element Vector{Int64}:
+ 5
+ 7
+ 9
+
+julia> df  # see that the previous expression updated the data frame `df`
+3×3 DataFrame
+ Row │ My First Column  My Second Column  My Third Column
+     │ Int64            Int64             Int64
 ─────┼────────────────────────────────────────────────────
-   1 │     0     67  male         2  own               67
-   2 │     1     22  female       2  own               67
-   3 │     2     49  male         1  own               67
-   4 │     3     45  male         2  free              67
-   5 │     4     53  male         2  free              67
-   6 │     5     35  male         1  free              67
-   7 │     6     53  male         2  own               67
-   8 │     7     35  male         3  rent              67
+   1 │               1                 4                5
+   2 │               2                 5                7
+   3 │               3                 6                9
 ```
 
-In the example below we are swapping values stored in columns `:Sex` and `:Age`:
+One benefit of using manipulation functions is that
+the name of the data frame only needs to be written once.
+(The `@with` macro from the
```

Comment:
Can you please remove the reference to `@with`. Here we should only teach DataFrames.jl. The point is that such references are super hard to maintain.

In particular soon `@with` will be removed from DataFramesMeta.jl, see https://github.com/JuliaData/DataFramesMeta.jl/issues/364, and then we would have an outdated documentation (as likely we will forget to change things here).

Reply:
Sure, makes sense.